x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/EnableSpatialDistancePushdown.java

```diff
@@ -44,23 +53,68 @@ public class EnableSpatialDistancePushdown extends PhysicalOptimizerRules.Parame
     @Override
     protected PhysicalPlan rule(FilterExec filterExec, LocalPhysicalOptimizerContext ctx) {
```

Comment:
Good point. I've added new content to the javadoc to describe the new scenario where we get the distance from an EVAL instead of directly within the WHERE clause.